tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet.md

```diff
@@ -0,0 +1,292 @@
+        - Emits a `CounterValueSet` event on success
+
+    ```rust
+    --8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet/lib.rs:75:99'
+    ```
+
+???+ function "increment(origin: OriginFor<T>, amount_to_increment: u32) -> DispatchResult"
+    This call increments the counter by a specified amount. It is accessible to any signed account.
+
+    - Parameters:
+        - `amount_to_increment` - the amount to add to the counter
+    - Validations:
+        - Prevents overflow during the addition
+        - Ensures the resulting counter value does not exceed `CounterMaxValue`
+    - Behavior:
+        - Updates the `CounterValue` storage item
+        - Tracks the number of interactions by the user in the `UserInteractions` storage map
+        - Emits a `CounterIncremented` event on success
+
+    ```rust
+    --8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet/lib.rs:101:141'
+    ```
+
+
+???+ function "decrement(origin: OriginFor<T>, amount_to_decrement: u32) -> DispatchResult"
+    This call decrements the counter by a specified amount. It is accessible to any signed account.
+
+    - Parameters:
+        - `amount_to_decrement` - the amount to subtract from the counter
+    - Validations:
+        - Prevents underflow during the subtraction
+        - Ensures the counter does not drop below zero
+    - Behavior:
+        - Updates the `CounterValue` storage item
+        - Tracks the number of interactions by the user in the `UserInteractions` storage map
+        - Emits a `CounterDecremented` event on success
+
+    ```rust
+    --8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet/lib.rs:143:178'
+    ```
+
+## Verify Compilation
+
+After implementing all the pallet components, verifying that the code still compiles successfully is crucial. Run the following command in your terminal to ensure there are no errors:
+
+```bash
+cargo build --package custom-pallet
+```
+
+If you encounter any errors or warnings, carefully review your code to resolve the issues. Once the build is complete without errors, your pallet implementation is ready.
+
+## Key Takeaways
+
+In this tutorial, you learned how to create a custom pallet by defining storage, implementing errors, adding dispatchable calls, and emitting events. These are the foundational building blocks for developing robust Polkadot SDK-based blockchain logic.
+
+To review this implementation, you can find the complete pallet code below:
+
+???+ example "Complete Pallet Code"
+    ```rust
+    --8<-- 'code/tutorials/polkadot-sdk/parachains/build-custom-pallet/build-pallet/lib.rs'
+    ```
+
+## Where to Go Next
+
+To ensure the reliability of your pallet, it’s crucial to test its functionality thoroughly. 
+
+Proceed to the following tutorial: [Pallet Testing](TODO: add-path), where you’ll learn how to write comprehensive tests for your pallet to validate its behavior and prevent bugs.
```

Comment:
We shouldn't add TODO:add-path anymore. I think a good file name for the next tutorial could be pallet-unit-testing.md because pallet-testing.md is already in use here: /develop/parachains/customize-parachain/pallet-testing/

Reply:
I'll just wait for the new structure before linking to the testing tutorial